deployment/ec2/amis.py

```diff
@@ -0,0 +1,35 @@
+"""Helper functions to handle EC2 related operations with Boto"""
+
+import boto
+import logging
+
+
+LOGGER = logging.getLogger('mmw')
+LOGGER.addHandler(logging.StreamHandler())
+LOGGER.setLevel(logging.INFO)
+
+
+def _prune_ami(ami):
+    """Actually deregister AMI and its associated snapshot"""
+    LOGGER.info('Identified that [%s] is eligible to be pruned..', ami.id)
+
+    ami.deregister(delete_snapshot=True)
+
+    LOGGER.info('Deregistered [%s] and snapshot [%s]..', ami.id,
+                ami.block_device_mapping['/dev/sda1'].snapshot_id)
+
+
+def prune(mmw_config, service, keep, aws_profile):
+    """Filter owned AMIs by service, environment, and count"""
+    stack_type = mmw_config['StackType']
+
+    conn = boto.connect_ec2(profile_name=aws_profile)
+    images = conn.get_all_images(owners='self', filters={
+        'tag:Service': service,
+        'tag:Environment': stack_type
+    })
+
+    if len(images) > keep:
+        map(_prune_ami, reversed(sorted(images[0:len(images) - keep])))
```

Comment:
If `keep` is 5 then this would remove the 5 oldest amis instead of keeping the 5 newest amis. May want to do something like this instead `list(reversed(sorted(images)))[0:keep]`?